adapters/mqtt-vertx-base/src/main/java/org/eclipse/hono/adapter/mqtt/AbstractVertxBasedMqttProtocolAdapter.java

```diff
@@ -1227,4 +1008,406 @@ private static MqttConnectReturnCode getConnectReturnCode(final Throwable e) {
             return MqttConnectReturnCode.CONNECTION_REFUSED_NOT_AUTHORIZED;
         }
     }
+
+
+    /**
+     * A class that contains handlers for command subscription, unsubscription and sending of commands.
+     */
+    protected class CommandHandler {
```

Comment:
The way this class is currently implemented, it is very tightly coupled to its enclosing class because it accesses many of its private fields and methods.
IMHO this is not very desirable because it makes it hard(er) to test. Maybe we can keep the OpenTracing related stuff and the logic for processing the MQTT messages in the enclosing class and use the `CommandHandler` only for keeping the state. WDYT?

Reply:
I will use the `CommandHandler `only for keeping the state (subscriptions and pubacks).  I will move back `onSubscribe`, `onUnsubscribe`, `onCommandReceived` and related methods to the enclosed class as before.

Reply:
It would probably be a good idea to factor out this class into its own file so that you are no longer tempted to tightly couple them together ;-)